input/input_driver.c

```diff
@@ -4486,8 +4486,28 @@ void joypad_driver_reinit(void *data, const char *joypad_driver_name)
  **/
 float input_get_sensor_state(unsigned port, unsigned id)
 {
-   bool input_sensors_enable              = config_get_ptr()->bools.input_sensors_enable;
-   return input_driver_get_sensor(port, input_sensors_enable, id);
+   settings_t *settings                   = config_get_ptr();
+   bool input_sensors_enable              = settings->bools.input_sensors_enable;
+   float sensitivity;
+   int remapped_port=settings->uints.input_sensor_index[port];
+   int invert;
+   unsigned remapped_input_sensor_id      = settings->uints.input_sensor_ids[port][id]/2;
```

Comment:
I just noticed this now, but if asking for `RETRO_SENSOR_ILLUMINANCE` (= 6) then this line results in undefined behavior because the second dimension of `input_sensor_ids` is too short. `RETROPAD_RETRO_SENSOR_LAST` is 6, so `unsigned input_sensor_ids[MAX_USERS][RETROPAD_RETRO_SENSOR_LAST]` hides an out-of-bounds access. Since light sensors aren't mapped to controllers the same way, you can probably special-case this whole function around them.

Reply:
Since this function is directly exposed to cores, it should validate the arguments against the available ports and IDs _before_ doing any lookup.